lms/djangoapps/branding/api.py

```diff
@@ -194,111 +200,128 @@ def _build_support_form_url():
-        ]
+        for link_name, (link_url, link_title) in links.items()
         if link_url and link_url != "#"
     ]
 
 
-def _footer_legal_links():
+def _footer_legal_links(language="en"):
     """Return the legal footer links (e.g. terms of service). """
 
     links = [
-        ("terms_of_service_and_honor_code", marketing_link("TOS_AND_HONOR"), _("Terms of Service & Honor Code")),
-        ("privacy_policy", marketing_link("PRIVACY"), _("Privacy Policy")),
-        ("accessibility_policy", marketing_link("ACCESSIBILITY"), _("Accessibility Policy")),
-        ("sitemap", marketing_link("SITE_MAP"), _("Sitemap")),
-        ("media_kit", marketing_link("MEDIA_KIT"), _("Media Kit")),
+        ("terms_of_service_and_honor_code", (marketing_link("TOS_AND_HONOR"), _("Terms of Service & Honor Code"))),
+        ("privacy_policy", (marketing_link("PRIVACY"), _("Privacy Policy"))),
+        ("accessibility_policy", (marketing_link("ACCESSIBILITY"), _("Accessibility Policy"))),
+        ("sitemap", (marketing_link("SITE_MAP"), _("Sitemap"))),
+        ("media_kit", (marketing_link("MEDIA_KIT"), _("Media Kit"))),
     ]
 
     # Backwards compatibility: If a combined "terms of service and honor code"
     # link isn't provided, add separate TOS and honor code links.
     tos_and_honor_link = marketing_link("TOS_AND_HONOR")
     if not (tos_and_honor_link and tos_and_honor_link != "#"):
         links.extend([
-            ("terms_of_service", marketing_link("TOS"), _("Terms of Service")),
-            ("honor_code", marketing_link("HONOR"), _("Honor Code")),
+            ("terms_of_service", (marketing_link("TOS"), _("Terms of Service"))),
+            ("honor_code", (marketing_link("HONOR"), _("Honor Code"))),
         ])
 
+    links = OrderedDict(links)
+    if language.startswith("es"):
+        del links["sitemap"]
+
     return [
         {
             "name": link_name,
             "title": link_title,
             "url": link_url,
         }
-        for link_name, link_url, link_title in links
+        for link_name, (link_url, link_title) in links.items()
         if link_url and link_url != "#"
     ]
 
 
-def _footer_business_links():
+def _footer_business_links(language="en"):
     """Return the business links to display in the footer. """
     platform_name = configuration_helpers.get_value('platform_name', settings.PLATFORM_NAME)
+    links = OrderedDict([
+        ("about", (marketing_link("ABOUT"), _("About"))),
+        ("enterprise", (marketing_link("ENTERPRISE"),
+         _("{platform_name} for Business").format(platform_name=platform_name))),
+        ("affiliates", (marketing_link("AFFILIATES"), _("Affiliates"))),
+        ("openedx", (_footer_openedx_link()["url"], _("Open edX"))),
+        ("careers", (marketing_link("CAREERS"), _("Careers"))),
+        ("news", (marketing_link("NEWS"), _("News"))),
+    ])
+
+    if language.startswith("es"):
+        del links["affiliates"]
```

Comment:
Having a delete function would be more convenient. You can pass array of keys and dict

Reply:
And here.